APIReplayDownloader/database_load.py

```diff
@@ -1,16 +1,109 @@
 import database
-from playerdata import get_leaderboard_by_page
+import transaction
+from playerdata import get_user_by_id
 from time import sleep
 
+# 4k section
 
-lb_db = database.open_db("db/leaderboard.fs")
-connection = database.get_connection(db)
+leaderboard_4k_db = database.open_db("db/leaderboard_4k.fs")
+leaderboard_4k_connection = database.get_connection(leaderboard_4k_db)
 
-for i in range(0, 71):
-    lb_page = get_leaderboard_by_page(i)
-    for user in lb_page:
-        print("Adding lb entry to database: ", database.insert_leaderboard_entry(connection, user))
+player_db = database.open_db("db/players.fs")
+player_connection = database.get_connection(player_db)
+
+print("Pre-packing database...")
+
+player_connection, player_db = database.pack_db(player_connection, player_db)
+
+print("Adding players from 4k leaderboard to database...")
+
+num_profiles_added = 0
+root = leaderboard_4k_connection.root()
+num_players = len(root.keys())
+
+for rank in root.keys():  # for all ranks on the 4k leaderboard
+    player_id = root[rank]["id"]
+
+    if num_profiles_added == 0:
+        transaction.begin()
+
+    if database.get_player(player_connection, player_id) != {}:
+        print(f"Player (id {player_id}) already in database. Skipping...")
+        continue
+
+    player_profile = get_user_by_id(player_id, 1)  # 1 means 4k
```

Comment:
another way to say what parameter values mean is to use variables like so
```
mode = 1
player_profile = get_user_by_id(player_id, mode)
```
this makes it explicit what it means. Its more easy than a comment when you have multiple values being fed in like `my_func(1,4,32,7)`